test-requirements.txt

```diff
@@ -5,7 +5,10 @@ black==23.3.0  # must match version in .pre-commit-config.yaml
 filelock>=3.3.0
 # lxml 4.9.3 switched to manylinux_2_28, the wheel builder still uses manylinux2014
 lxml>=4.9.1,<4.9.3; (python_version<'3.11' or sys_platform!='win32') and python_version<'3.12'
-pre-commit
+# PyYAML 6.0 (required by pre-commit) can't be build with cython 3.0.0
+# There aren't any wheels for Python 3.12 yet.
+# https://github.com/yaml/pyyaml/issues/724
+pre-commit; python_version<'3.12'
```

Comment:
Seems like the 32-bit CI is failing on `master` with Python <3.12: https://github.com/python/mypy/actions/runs/5579778257/jobs/10195786014

Presumably that's because PyYAML doesn't provide wheels for 32-bit platforms, so PyYAML is being built from source in that CI job as well?

Reply:
Hmm, the `platform_machine!='i386'` specifier doesn't seem to work 🤔

Reply:
It seems that `platform.machine()` and the `platform_machine` value used by `pip` don't match up. The job itself is running on Ubuntu `64bit`, so `machine()` is `x86_64`. However Python itself is build for `i686` which is used by `pip`.